lib/src/Uncertainty/Distribution/DistFunc.cxx

```diff
@@ -1958,6 +1958,69 @@ Scalar DistFunc::qDickeyFullerNoConstant(const Scalar p,
   throw NotYetImplementedException(HERE) << "In DistFunc::qDickeyFullerNoConstant(const Scalar p, const Bool tail): cannot give quantile value for the level " << p << ". Value is missing in table";
 }
 
+// For uniform distribution over a segment
+Point DistFunc::rUniformSegment(const Point & a,
+                                const Point & b)
+{
+  const UnsignedInteger dimension = a.getDimension();
+  if (b.getDimension() != dimension) throw InvalidArgumentException(HERE) << "Error: the second point has a dimension=" << b.getDimension() << ", expected dimension=" << dimension;
+  const Scalar u = RandomGenerator::Generate();
+  return u * a + (1.0 - u) * b;
+}
+
+Sample DistFunc::rUniformSegment(const Point & a,
+                                 const Point & b,
+                                 const UnsignedInteger size)
+{
+  const UnsignedInteger dimension = a.getDimension();
+  if (b.getDimension() != dimension) throw InvalidArgumentException(HERE) << "Error: the second point has a dimension=" << b.getDimension() << ", expected dimension=" << dimension;
+  Sample result(size, dimension);
+  for (UnsignedInteger n = 0; n < size; ++n)
+  {
+    const Scalar u = RandomGenerator::Generate();
```

Comment:
There is no actual advantage in generating the pseudo-random numbers in advance rather than on the fly if:
 * you don't apply an expensive transformation on these numbers
 * you don't apply it in parallel

The way modern pseudo-random numbers generators work is to manage a huge state, slice bits in it to generate unitary realizations then perform a big update once the current state has been exhausted.